Hardware/Rpi4.md

```diff
@@ -15,82 +15,76 @@ SPDX-License-Identifier: CC-BY-SA-4.0
 cd u-boot
-git revert 995eab8b5b580b67394312b1621c60a71042cd18
 make CROSS_COMPILE=aarch64-linux-gnu- rpi_4_defconfig
 make CROSS_COMPILE=aarch64-linux-gnu-
 ```
 
+More information regarding U-Boot support for the Raspberry Pi can be found [here](https://u-boot.readthedocs.io/en/latest/board/broadcom/raspberrypi.html).
+
 This will enable you to get the most up-to-date U-boot which will boot
 seL4 on the RPi4 successfully.
 
-**NOTE:** Automatic revert is not going to work if you are using the
-latest version of u-boot. You need to manually revert the change by
-looking at the changeset.
-
-
 ## SD card setup
- The PI boots from the first FAT32 partition on the
-SD card. Where files are specified, they should be located in the root
-directory of this partition.
+The RPi4 boots from the first FAT32 partition on the microSD card. Where files are specified,
+they should be located in the root directory of this partition.
 
 |Stage |Filename |Description |Source|
 |-|-|-|-|
 |FSBL |- |Mounts SD and loads SSBL |ROM |
 |SSBL |bootcode.bin|Loads GPU firmware and boots GPU|<https://github.com/raspberrypi/firmware/tree/master/boot> |
 |GPU firmware |start4.elf |Loads CPU bootloader and boots CPU |<https://github.com/raspberrypi/firmware/tree/master/boot> |
-|Usually the Linux kernel, but could also be u-boot |u-boot.bin |u-boot| Compiled using the instructions above |
-||config.txt|u-boot parameters |Add `arm_64bit=1`, `kernel=u-boot.bin`, and `dtoverlay=disable-bt` to the bottom of config.txt |
-||uboot.env |u-boot saved environment |Generated by u-boot (default environment) bootcmd copied to bootcmd_orig bootcmd and bootdelay removed |
+|Usually the Linux kernel, but could also be U-Boot |u-boot.bin |U-Boot binary| Compiled using the instructions above |
+||config.txt|U-Boot parameters |Add `arm_64bit=1`, `kernel=u-boot.bin`, and `dtoverlay=disable-bt` to the bottom of `config.txt` |
+||uboot.env |U-Boot saved environment |Generated by U-Boot (default environment) bootcmd copied to bootcmd_orig bootcmd and bootdelay removed. This file will not exist when you first setup your SD card. |
 ||bcm2711-rpi-4-b.dtb | RPi4 device tree blob |<https://github.com/raspberrypi/firmware/tree/master/boot> |
 ||overlays/* | Device tree overlays |<https://github.com/raspberrypi/firmware/tree/master/boot/overlays> |
 
 ## Getting seL4 onto your Raspberry Pi 4
 
-Because you have applied a device tree overlay to use the UART on GPIO pins 14 and 15,
-you will have to update seL4 to use serial0 instead of serial1.
-In the kernel at `src/plat/bcm2711/overlay-rpi4.dts`, you should replace all instances of
-`serial1` with `serial0`.
+The final step to test that you have setup your RPi4 correctly is to build and run the
+seL4test project.
 
-Then, you should then build the kernel, such as in seL4test:
+You will see that the `init-build.sh` command contains the flag
+`-DRPI4_MEMORY`, this specifies how much RAM exists on the Raspberry Pi that will run
```

Comment:
We should also mention 8 GiB used as default.

Reply:
Done